server/dataset_context.cc

```diff
@@ -75,10 +75,10 @@ doGetFunction(const Utf8String & tableName,
                     }
                 };
 
-            BoundFunction boundFunction(exec, applier->info.output);
-            boundFunction.resultMetadata.isConstant = constantArgs && applier->info.deterministic;
+            bool isConst = constantArgs && applier->info.deterministic;
+            auto outputInfo = applier->info.output->getConst(isConst);
```

Comment:
I am not sure why we have a getConst function.  I would set constness in constructor and have a setConst member function, that is:
```
applier->info.output.setConst(constantArgs && applier->info.deterministic);
return BoundFunction(exec, applier->info.output)
```